### PR TITLE
Scripts for iOS size audit

### PR DESCRIPTION
**Pull request by muxi**
The scripts for building `Sample` example app and compare sizes with another commit. Configuration of Kokoro to use this script must be submitted in another PR; to be followed up.

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

File: tools/profiling/ios_bin/binary_diff.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+from parse_link_map import parse_link_map
+
+sys.path.append(
+    os.path.join(
+        os.path.dirname(sys.argv[0]), '..', '..', 'run_tests', 'python_utils'))
+import comment_on_pr
+
+size_labels = ('Core', 'ObjC', 'BoringSSL', 'Protobuf', 'Total')
+
+argp = argparse.ArgumentParser(description='Perform diff on microbenchmarks')
```
Comment (ncteisen) [inline tools/profiling/ios_bin/binary_diff.py]: Update description

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by ncteisen** [comment]
I would suggest adding some logic that only prints rows if there is a delta. I know if doesn't seem like much, but asking the developer to scan and look for a difference makes them less likely to actually see the data.

Notice the other bots print "No significant performance differences" or something similar if there was no change

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by muxi** [comment]
Thanks for looking at it, Noah. A sample output is like below:
```
******************STATIC*******************
  New size                      Old size
   673,670           Core        673,670
    78,291           ObjC         78,291
   725,783      BoringSSL        725,783
   445,326       Protobuf        445,326

 1,994,105          Total      1,994,105

****************FRAMEWORKS*****************
  New size                      Old size
 3,511,578           Core      3,511,578
 1,192,497           ObjC      1,192,497
 3,211,300      BoringSSL      3,211,300
 2,003,673       Protobuf      2,003,673

10,589,293          Total     10,589,285
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by muxi** [comment]
New sample:
```
******************STATIC*******************
  New size                      Old size
   673,670       Core (=)        673,670
    78,291       ObjC (=)         78,291
   725,783  BoringSSL (=)        725,783
   445,326   Protobuf (=)        445,326

 1,994,105      Total (=)      1,994,105

****************FRAMEWORKS*****************
  New size                      Old size
 3,511,578       Core (=)      3,511,578
 1,192,497       ObjC (=)      1,192,497
 3,211,300  BoringSSL (=)      3,211,300
 2,003,673   Protobuf (=)      2,003,673

10,589,295      Total (>)     10,589,293
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by muxi** [comment]
Bazel build for C/C++ - #15610 
Interop Cloud-to-Prod Tests - github issue

**Pull request merged**